mapper-processor/src/test/java/com/datastax/oss/driver/internal/mapper/processor/DaoAnnotationTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.mapper.processor;
+
+import static com.google.testing.compile.CompilationSubject.assertThat;
+
+import com.datastax.oss.driver.api.mapper.annotations.Dao;
+import com.google.testing.compile.Compilation;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.TypeSpec;
+import javax.lang.model.element.Modifier;
+import javax.tools.StandardLocation;
+import org.junit.Test;
+
+public class DaoAnnotationTest extends MapperProcessorTest {
+
+  @Test
+  public void should_work_on_nested_interface() {
+    Compilation compilation =
+        compileWithMapperProcessor(
+            "test",
+            TypeSpec.classBuilder(ClassName.get("test", "Foo"))
+                .addModifiers(Modifier.PUBLIC)
+                .addType(
+                    TypeSpec.interfaceBuilder("Bar")
+                        .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+                        .addAnnotation(Dao.class)
+                        .build())
+                .build());
```

Comment:
JavaPoet works pretty well to embed source code directly in the test. It is a bit more refactoring-friendly than raw strings (for example if we rename annotations).

I like that the test is self-contained (no external source files): it's easier to read, also it lowers the bar to add more tests (no need to think of names for the new files, etc).

Reply:
I agree that embedding source code in a test makes it better for maintenance. 
But on the other hand, if our users that have no previous experience with annotation processors will want to understand how to use mapper they may want to go to tests to see how it work.
If they don't know javaPoet that test may be really hard to understand for them. If they will have raw `Foo.java` file used by the test that will be easier for them to reason about the behaviour

Reply:
I wouldn't consider those unit tests as examples. The integration tests are a better fit for that IMO (and we'll definitely also cover the mapper in examples / manual).